### ISSUE TITLE
libOTe/cryptotools needs to have ENABLE_CIRCUITS defined.

### ISSUE THREAD

**Issue by 1walktheline** [ladnir/aby3#31]
When I run "python3 build.py", I get the error below.

Cmake Error at CMakeLists.txt: 72(message):
    libOTe/cryptotools needs to have ENABLE_CIRCUITS defined

This error occurs in both Windows and Ubuntu, but when I tried to change the latest updade back(I mean change line 51 from "-DENABLE_CIRCUITS" back to "-- -DENABLE_CIRCUITS" in the file thirdparty/getLibOTe.py), the Error disappeared and the process of build succeed.

I'm wondering why to make this update(from "-- -D" to "-D") and whether it is necessary? Thank you very much!

**Comment by ladnir** [comment]
libOTe has been updated and no longer expects to see `--`. You likely just did a `git pull` on aby3 and this caused libOTe to be out of sync (libOTe is not pulled once it is found). You need to update libOTe. You can do this by deleting it `out/install/<platform>/lib/cmake/libOTe/`

**Comment by ladnir** [comment]
It's a good idea to always take a dependency to a specific commit as opposed to just `git clone aby3`.

**Comment by 1walktheline** [comment]
Oh, I see. Thank you so much!

**Comment by 1walktheline** [comment]
Unfortunately, when I build aby3 on a new ubuntu20.04 system, the same problem occurred again and I have no idea how to solve it. The code is cloned from GitHub today. Do you have  any suggestion? 
The complete output is:

$ sudo python3 build.py
======= build.py (aby3) ==========
mkdir -p out/build/linux
cmake -S . -B out/build/linux  -DCMAKE_BUILD_TYPE=Release
cmake --build out/build/linux   --parallel 16 
vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv

-- Boost_LIB: Boost::system;Boost::thread
-- Boost_INC: /home/ivy/aby3/aby3/thirdparty/unix/include


CMake Error at CMakeLists.txt:72 (message):
  libOTe/cryptoTools needs to have ENABLE_CIRCUITS defined.


-- Configuring incomplete, errors occurred!
See also "/home/ivy/aby3/aby3/out/build/linux/CMakeFiles/CMakeOutput.log".
make: Makefile: 没有那个文件或目录
make: *** 没有规则可制作目标“Makefile”。 停止。

**Comment by 1walktheline** [comment]
Later on when I built aby3 on a new Ubuntu OS this problem occurred again and discarding the latest update in aby3 did not matter this time. I realized that it was not the update that resulted in the error. I found the right solution and record it as following.

Change directory into: 
cd aby3/thirdparty/libOTe/cryptoTools

Modify CMakeLists.txt to set the default value of ENABLE_CIRCUITS ON:
vim CMakeLists.txt

In line 56, exchange 
"option(ENABLE_CIRCUITS  "compile the circuit module" OFF)"
with 
"option(ENABLE_CIRCUITS  "compile the circuit module" ON)"
This makes sense.

**Comment by 1walktheline** [comment]
Solution record.

Later on when I built aby3 on a new Ubuntu OS this problem occurred again and discarding the latest update in aby3 did not matter this time. I realized that it was not the update that resulted in the error. I found the right solution and record it as following.

Change directory into:
cd aby3/thirdparty/libOTe/cryptoTools

Modify CMakeLists.txt to set the default value of ENABLE_CIRCUITS ON:
vim CMakeLists.txt

In line 56, exchange
"option(ENABLE_CIRCUITS "compile the circuit module" OFF)"
with
"option(ENABLE_CIRCUITS "compile the circuit module" ON)"
This makes sense.
